src/controllers/account/index.ts

```diff
@@ -1,45 +1,68 @@
-    // const emailToken = await AccountUtilities.getEmailResetToken(resetToken)
-    // // TODO email token to user
+    ctx.assert(email, 400, Strings.EmailIsRequired)
 
-    ctx.status = 200
-    ctx.body = { ok: true }
-  }
-  public static async receiveReset (ctx: BaseContext) {
-    const { token } = ctx.request.body
-    ctx.assert(token, 400)
-    // const resetToken = await AccountUtilities.splitSelectorVerifier(token)
-    // DB lookup first 16 bytes provided (selector)
-    // if found, hash next 16 bytes provided (verifier)
-    // const verifierHash = await AccountUtilities.getVerifierHash(resetToken)
-    // compare hash with verifier hash in DB
+    // Generate token, store it
+    const emailToken = await generateAndStoreToken(ctx, false)
     try {
-      // const storedVerifierHash = await AccountUtilities.getStoredVerifierHash(resetToken)
-      // const match = await AccountUtilities.compareHashes(storedVerifierHash, verifierHash)
-      // if (!match) {
-      //   // if hash does not match, 401 and delete token
-      //   await AccountUtilities.deleteSelector(resetToken)
-      //   ctx.assert(match, 401)
-      // }
+      await Methods.sendMagicLinkEmail(email, emailToken)
+      ctx.body = { ok: true }
     } catch (e) {
-      ctx.throw(401)
+      console.log(e)
+      ctx.body = { ok: false, message: e.message || 'unknown error' }
     }
+  }
+
+  // Endpoint to create hash to send in email to the user on register
+  public static async register (ctx: BaseContext) {
+    const { email, role } = ctx.request.body
+    ctx.assert(email, 400, Strings.EmailIsRequired)
+    ctx.assert(role, 400, Strings.RoleIsRequired)
 
-    // if match and hasn't expired, reset the password
-    return AccountController.resetPassword(ctx)
+    // Generate token, store it
+    const emailToken = await generateAndStoreToken(ctx, false)
+    try {
+      await Methods.sendMagicLinkEmail(email, emailToken)
+      ctx.body = { ok: true }
+    } catch (e) {
+      ctx.body = { ok: false, message: e.message || 'unkonwn error' }
+    }
   }
-  public static async resetPassword (ctx: BaseContext) {
-    ctx.status = 200
-    ctx.body = { ok: true }
+
+  // Endpoint to verify hash from email magic link and send back auth token
+  public static async verifyAuth (ctx: BaseContext) {
+    const { email, auth } = ctx.request.body
+    ctx.assert(email, 400, Strings.EmailIsRequired)
+    ctx.assert(auth, 400, Strings.AuthIsRequired)
+
+    // Look up hash sent in request in the db 
+    try {
+      const token = await getStoredVerifierHash(ctx.request.body.auth)
+      // If email matches the tokens email, they're authd and we should replace 
+      // token with long lasting token and respond with that token + an ok status
+      if (token.email === email) {
+        
+        // Generate long live token, store it
+        const longLiveToken = await generateAndStoreToken(ctx, true)
```

Comment:
This works well for login verification; in the case of registration verification, this is the point where we would create the new user in MySQL. We could potentially then continue with the login logic so they don't have to go through another magic link dance to login after verifying registration.

Reply:
my understanding is at this point we're just sending the long live token up to the client to use in all auth'd reqs, so no more dance after this point